nix/check-nix-tools.nix

```diff
@@ -0,0 +1,76 @@
+# This is a a CI check script that runs nix/regenerate.sh in the
+# project repository. If anything changes, then it uploads the patch
+# to Buildkite. If run on a PR branch, and there is a SSH key present,
+# it will attempt to push the changes back to the PR.
+
+{ stdenv, writeScript, coreutils, nixStable, git, gawk }:
+
+with stdenv.lib;
+
+writeScript "check-nix-tools.sh" ''
+  #!${stdenv.shell}
+
+  set -euo pipefail
+
+  export PATH="${makeBinPath [ stdenv.shellPackage coreutils nixStable git gawk ]}:$PATH"
+
+  cd $(git rev-parse --show-toplevel)
+
+  # The regenerate script is here by convention
+  ./nix/regenerate.sh
+
+  # The generated files will appear somewhere under ./nix
+  git add -A nix
+
+  # Check if there are changes staged for commit.
+  if git diff-index --cached --quiet HEAD --; then
+    echo "Generated Nix code is up-to-date."
+    exit 0
+  else
+    echo "Committing changes..."
+    commit_message="Regenerate nix"
+
+    # If on a PR branch, search for a previous regen commit to fix up.
+    commit_fixup=""
+    if [ -n "''${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-}" ]; then
+      previous=$(git log --pretty=oneline --no-decorate origin/$BUILDKITE_PULL_REQUEST_BASE_BRANCH..HEAD | awk "/$commit_message/ { print \$1; }")
+      if [ -n $previous ]; then
+        commit_fixup="--fixup $previous"
+      fi
+    fi
+
+    # Create the commit
+    export GIT_COMMITTER_NAME="IOHK"
+    export GIT_COMMITTER_EMAIL="devops+nix-tools@iohk.io"
+    export GIT_AUTHOR_NAME="$GIT_COMMITTER_NAME"
+    export GIT_AUTHOR_EMAIL="$GIT_COMMITTER_EMAIL"
+    git commit --no-gpg-sign $commit_fixup --message "$commit_message"
```

Comment:
We do at the moment requires commit to be signed. Perhaps it possible to fill an exception for `devops+nix-tools@iohk.io`.

Reply:
Commit signing is possible. It means deploying a GPG key to the buildkite agents and then adding the key to a GitHub user such as @iohk-devops.